### PR TITLE
MB-15059 - Add review weights playwright test

### PR DESCRIPTION
**Pull request by michaelvaldes-truss**
## Summary

This PR should add a new playwright test for an excess weight scenario in which a move has excess weight, and a services counselor handles the alert by decreasing the amount of weight on a PPM weight ticket.

My goals for the test were the following:
1. Match the flow of the test to the flow an actual user might take
2. Don't deal with numerous edge cases -- leave that to the unit tests
3. Address the major points of the related user stories

Related User Stories
- SC can see move weights on the review-shipment-weights page [MB-14965](https://dp3.atlassian.net/browse/MB-14965)
- SC can see excess weight alerts for a move with excess weight [MB-14968](https://dp3.atlassian.net/browse/MB-14968)
- SC can adjust PPM weight tickets to address excess weight [MB-12170](https://dp3.atlassian.net/browse/MB-12170)

[this article](https://transcom.github.io/mymove-docs/docs/frontend/testing/writing-playwright-tests-for-milmove/) explains more about the approach used.

## Setup to Run Your Code

<details>
<summary>💻 You will need to use one terminal to test this locally.</summary>

##### Terminal 1

Run the playwright test

```sh
npx playwright test playwright/tests/office/servicescounseling/servicesCounselingWeightTickets.spec.js
```

</details>

## Verification Steps for Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Request review from a member of a different team.
- [ ] Have the Jira acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Frontend

- [ ] User facing changes have been reviewed by design.
- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/main/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
- [ ] There are no new console errors in the browser devtools
- [ ] There are no new console errors in the test output
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.

[MB-14965]: https://dp3.atlassian.net/browse/MB-14965?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[MB-14968]: https://dp3.atlassian.net/browse/MB-14968?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[MB-12170]: https://dp3.atlassian.net/browse/MB-12170?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: playwright/tests/office/servicescounseling/servicesCounselingWeightTickets.spec.js [hunk 1]
```diff
@@ -38,3 +38,38 @@ test('A service counselor can approve/reject weight tickets', async ({ page, scP
...
+  await page.getByTestId('formRemarks').fill('Reduce PPM weight to slash excess weight');
+  await page.getByRole('button', { name: 'Save changes' }).click();
+  await page.getByRole('button', { name: 'Continue' }).click();
+  await page.getByTestId('closeSidebar').click();
+
+  // navigate to review-shipment-weights page and verify excess weight alert is no longer visible
+  await page.getByRole('button', { name: 'Review shipment weights' }).click();
+  expect(
+    await page.getByText('This move has excess weight. Review PPM weight ticket documents to resolve.').count(),
+  ).toEqual(0);
```
Comment (ruizajtruss) [inline playwright/tests/office/servicescounseling/servicesCounselingWeightTickets.spec.js:74]: Suggestion: Could simplify this a tiny bit with `toHaveCount` and it's more inline with the playwright docs you linked in the PR too
```suggestion
    await expect(page.getByText('This move has excess weight. Review PPM weight ticket documents to resolve.')).toHaveCount(0);
```

File: playwright/tests/office/servicescounseling/servicesCounselingWeightTickets.spec.js [hunk 2]
```diff
@@ -38,3 +38,38 @@ test('A service counselor can approve/reject weight tickets', async ({ page, scP
...
+  await expect(page.getByText('Weight moved', { exact: true })).toBeVisible();
+
+  // verify that the excess weight alert is visible, since the move has excess weight
+  await expect(
+    page.getByText('This move has excess weight. Review PPM weight ticket documents to resolve.'),
+  ).toBeVisible();
+
+  // navigate to review-documents page and decrease ppm shipment weight below threshold
+  await page.getByRole('link', { name: 'Review Documents' }).click();
+  // await page.getByRole('heading', { name: 'Edit PPM net weight' }).click();
```
Comment (ruizajtruss) [inline playwright/tests/office/servicescounseling/servicesCounselingWeightTickets.spec.js:62]: Can remove this commented out line

File: playwright/tests/office/servicescounseling/servicesCounselingWeightTickets.spec.js [hunk 3]
```diff
@@ -38,3 +38,38 @@ test('A service counselor can approve/reject weight tickets', async ({ page, scP
...
+  await scPage.navigateToCloseoutMove(move.locator);
+
+  // navigate to review-shipment-weights page and verify page components are rendered
+  await page.getByRole('button', { name: 'Review shipment weights' }).click();
+  await expect(page.getByText('Weight allowance', { exact: true })).toBeVisible();
+  await expect(page.getByText('Estimated weight (total)', { exact: true })).toBeVisible();
+  await expect(page.getByText('Max billable weight', { exact: true })).toBeVisible();
+  await expect(page.getByText('Move weight (total)', { exact: true })).toBeVisible();
+  await expect(page.getByText('Weight moved by customer', { exact: true })).toBeVisible();
+  await expect(page.getByText('Weight moved', { exact: true })).toBeVisible();
```
Comment (ruizajtruss) [inline playwright/tests/office/servicescounseling/servicesCounselingWeightTickets.spec.js:53]: **Suggestion:** I was thinking it might be a good idea to include visibility checks for the column names that appear in these two sections similar to what you did above - but maybe that's too granular for these tests 🤔

**Review by ruizajtruss — CHANGES_REQUESTED** [review]
Couple of small suggestions and one change, other than that LGTM! Great job on this and I really appreciate how you included comments in your tests to break everything up to make it more readable/easier to quickly understand  🎉

File: playwright/tests/office/servicescounseling/servicesCounselingWeightTickets.spec.js [hunk 4]
```diff
@@ -38,3 +38,38 @@ test('A service counselor can approve/reject weight tickets', async ({ page, scP
   await expect(page.getByRole('radio', { name: 'Reject' })).toBeChecked();
   await expect(page.getByLabel('Reason')).toHaveValue('Justification for rejection');
 });
+
+test('A services counselor can reduce PPM weights for a move with excess weight', async ({ page, scPage }) => {
+  const move = await scPage.testHarness.buildApprovedMoveWithPPMShipmentAndExcessWeight();
+  await scPage.navigateToCloseoutMove(move.locator);
+
+  // navigate to review-shipment-weights page and verify page components are rendered
+  await page.getByRole('button', { name: 'Review shipment weights' }).click();
```
Comment (ahobson) [inline playwright/tests/office/servicescounseling/servicesCounselingWeightTickets.spec.js:47]: Ideally I think you would use or add a call to `scPage.waitForPage.something()` here to ensure the page is loaded after the navigation. In general, after clicking a button you want to wait for the new page before continuing.

File: playwright/tests/office/servicescounseling/servicesCounselingWeightTickets.spec.js [hunk 5]
```diff
@@ -38,3 +38,38 @@ test('A service counselor can approve/reject weight tickets', async ({ page, scP
...
+  await expect(page.getByText('Weight moved by customer', { exact: true })).toBeVisible();
+  await expect(page.getByText('Weight moved', { exact: true })).toBeVisible();
+
+  // verify that the excess weight alert is visible, since the move has excess weight
+  await expect(
+    page.getByText('This move has excess weight. Review PPM weight ticket documents to resolve.'),
+  ).toBeVisible();
+
+  // navigate to review-documents page and decrease ppm shipment weight below threshold
+  await page.getByRole('link', { name: 'Review Documents' }).click();
```
Comment (ahobson) [inline playwright/tests/office/servicescounseling/servicesCounselingWeightTickets.spec.js:61]: I think you should add ` await scPage.waitForPage.reviewDocuments();`here

File: playwright/tests/office/servicescounseling/servicesCounselingWeightTickets.spec.js [hunk 6]
```diff
@@ -38,3 +38,38 @@ test('A service counselor can approve/reject weight tickets', async ({ page, scP
...
+  // await page.getByRole('heading', { name: 'Edit PPM net weight' }).click();
+  await page.getByTestId('net-weight-display').getByTestId('button').click();
+  await page.getByTestId('weightInput').fill('0');
+  await page.getByTestId('formRemarks').fill('Reduce PPM weight to slash excess weight');
+  await page.getByRole('button', { name: 'Save changes' }).click();
+  await page.getByRole('button', { name: 'Continue' }).click();
+  await page.getByTestId('closeSidebar').click();
+
+  // navigate to review-shipment-weights page and verify excess weight alert is no longer visible
+  await page.getByRole('button', { name: 'Review shipment weights' }).click();
```
Comment (ahobson) [inline playwright/tests/office/servicescounseling/servicesCounselingWeightTickets.spec.js:71]: I think this is another opportunity for `waitForPage`

File: playwright/tests/office/servicescounseling/servicesCounselingWeightTickets.spec.js (hunk 3 above)
Comment (michaelvaldes-truss) [reply playwright/tests/office/servicescounseling/servicesCounselingWeightTickets.spec.js:55]: Yeah, I didn't want to end up checking everything on the page, but I don't feel super strongly about this either way.

**Review by ahobson — APPROVED** [review]
Awesome! Great Work

File: playwright/tests/office/servicescounseling/servicesCounselingWeightTickets.spec.js (hunk 3 above)
Comment (donaldthai) [reply playwright/tests/office/servicescounseling/servicesCounselingWeightTickets.spec.js:55]: > **Suggestion:** I was thinking it might be a good idea to include visibility checks for the column names that appear in these two sections similar to what you did above - but maybe that's too granular for these tests 🤔

Comment: I do feel like E2E should be used for testing the overall integration and flows. Unless there is a column that redirects somewhere, I would like to keep the granular checks like the column names to be on the unit test level.

**Review by ruizajtruss — APPROVED** [review]
